test/sqllogictest/type-promotion.slt

```diff
@@ -63,6 +63,39 @@ SELECT 'true' OR 'false'
 ----
 true
 
+# Int literals less than i32::MAX/MIN are int4
+query T
+SELECT pg_typeof(100)
+----
+integer
+
+query T
+SELECT pg_typeof(-100)
+----
+integer
+
+# Int literals that exceed i32, are int8
+query T
+SELECT pg_typeof(2147483648)
```

Comment:
For mz and `i32::MIN` this still gives the wrong result.

MZ:
```
materialize=> SELECT pg_typeof(-2147483648);
 pg_typeof
-----------
 bigint
(1 row)
```

Postgres:
```
aljoscha=# SELECT pg_typeof(-2147483648);
 pg_typeof
-----------
 integer
(1 row)
```

The problem seems to be that the literal is parsed separately from the minus, i.e. `-2147483648` is parsed as `(Negate (Literal 2147483648)` and `2147483648` is in fact too big for `i32`.

Reply:
Ah, there is code in the parser that tied the `-` sign to the literal that was unreachable! I didn't realize the code wasn't being reached, so many thanks for pointing out that gap. Should be fixed in the latest commit.